Crash on HarmonyOS 3(base on android 12)

### Checklist

- [X] I have used the search function for [open](https://github.com/AntennaPod/AntennaPod/issues) **and** [closed](https://github.com/AntennaPod/AntennaPod/issues?q=is%3Aissue+is%3Aclosed) issues to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain podcast, I will include the URL of that podcast.

### App version

3.0.0-beta6

### Where did you get the app from

Google Play

### Android version

12

### Device model

Honor V30 Pro

### First occurred

about 3 days ago

### Steps to reproduce

_No response_

### Expected behaviour

_No response_

### Current behaviour

Unable to start on the phone, when I click the icon ,it just crashes.

### Logs

03-06 18:27:13.066 18048 18048 E AndroidRuntime: FATAL EXCEPTION: main
03-06 18:27:13.066 18048 18048 E AndroidRuntime: Process: de.danoeh.antennapod, PID: 18048
03-06 18:27:13.066 18048 18048 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{de.danoeh.antennapod/de.danoeh.antennapod.activity.MainActivity}: java.lang.RuntimeException: com.google.android.gms.cast.framework.ModuleUnavailableException: com.google.android.gms.dynamite.DynamiteModule$LoadingException: Remote load failed. No local fallback found.
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4631)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4842)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:113)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2928)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:117)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:293)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:9596)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1204)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: Caused by: java.lang.RuntimeException: com.google.android.gms.cast.framework.ModuleUnavailableException: com.google.android.gms.dynamite.DynamiteModule$LoadingException: Remote load failed. No local fallback found.
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.cast.framework.CastContext.getSharedInstance(SourceFile:11)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at de.danoeh.antennapod.playback.cast.CastEnabledActivity.onCreate(SourceFile:23)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at de.danoeh.antennapod.activity.MainActivity.onCreate(SourceFile:111)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8488)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8461)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1343)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4602)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	... 12 more
03-06 18:27:13.066 18048 18048 E AndroidRuntime: Caused by: com.google.android.gms.cast.framework.ModuleUnavailableException: com.google.android.gms.dynamite.DynamiteModule$LoadingException: Remote load failed. No local fallback found.
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.internal.cast.zzad.zzf(SourceFile:7)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.internal.cast.zzad.zza(SourceFile:1)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.cast.framework.CastContext.<init>(SourceFile:4)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.cast.framework.CastContext.getSharedInstance(SourceFile:10)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	... 18 more
03-06 18:27:13.066 18048 18048 E AndroidRuntime: Caused by: com.google.android.gms.dynamite.DynamiteModule$LoadingException: Remote load failed. No local fallback found.
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.dynamite.DynamiteModule.load(SourceFile:58)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.internal.cast.zzad.zzf(SourceFile:2)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	... 21 more
03-06 18:27:13.066 18048 18048 E AndroidRuntime: Caused by: com.google.android.gms.dynamite.DynamiteModule$LoadingException: Failed to load remote module.
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	at com.google.android.gms.dynamite.DynamiteModule.load(SourceFile:27)
03-06 18:27:13.066 18048 18048 E AndroidRuntime: 	... 22 more

--- a/playback/cast/src/play/java/de/danoeh/antennapod/playback/cast/CastEnabledActivity.java
+++ b/playback/cast/src/play/java/de/danoeh/antennapod/playback/cast/CastEnabledActivity.java
@@ -20,7 +20,12 @@ public abstract class CastEnabledActivity extends AppCompatActivity {
         super.onCreate(savedInstanceState);
         canCast = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(this) == ConnectionResult.SUCCESS;
         if (canCast) {
-            CastContext.getSharedInstance(this);
+            try {
+                CastContext.getSharedInstance(this);
+            } catch (Exception e) {
+                e.printStackTrace();
+                canCast = false;
+            }
         }
     }
 

